src/plugins/discover/public/application/main/components/sidebar/discover_field.tsx

```diff
@@ -58,10 +58,19 @@ const FieldInfoIcon: React.FC = memo(() => (
   </EuiToolTip>
 ));
 
+/**
+ * Extracts the type from a data view field that will match the right icon.
+ *
+ * We define custom logic for Discover in order to distinguish between various "string" types.
+ */
+export const getTypeForFieldIcon = (field: DataViewField) =>
```

Comment:
Could you extract this new utility function to `/src/plugins/discover/public/utils`, and add a test, that would be awesome 👍